scripts/zones/Bibiki_Bay/npcs/qm_shen.lua

```diff
@@ -5,18 +5,13 @@
 -- !pos -115.108 0.300 -724.664 4
 -----------------------------------
 local ID = require("scripts/zones/Bibiki_Bay/IDs")
+require('scripts/globals/npc_util')
 -----------------------------------
 local entity = {}
 
 entity.onTrade = function(player, npc, trade)
-    if (not GetMobByID(ID.mob.SHEN):isSpawned() and trade:hasItemQty(1823, 1) and trade:getItemCount() == 1) then
-        player:tradeComplete()
-        SpawnMob(ID.mob.SHEN):updateClaim(player)
-        for i = 1, 2 do
-            if (not GetMobByID(ID.mob.SHEN+i):isSpawned()) then
-                SpawnMob(ID.mob.SHEN+i):updateEnmity(player)
-            end
-        end
+    if npcUtil.tradeHasExactly(trade, xi.items.SHRIMP_LANTERN) and npcUtil.popFromQM(player, npc, ID.mob.SHEN) then
+        player:confirmTrade()
```

Comment:
I'm pretty sure there's some messages with this interaction point
Found them, yeah!:
7618: Innumerable shrimp shells are floating in the water here.<7F31>
7619: The <01052382808080> begins skimming quietly across the surface of the water.\x07...What's this!?\x07An enormous creature is chasing after it from below!!!<7F31>

Reply:
Nice, didn't think to check that for this one.